### ISSUE TITLE
Problems with manifest

### ISSUE THREAD

**Issue by alexlarsson** [flathub/org.subsurface_divelog.Subsurface#2]
We had to fix something due to a change in the kde runtime, and the build failed:
https://flathub.org/builds/#/builders/4/builds/967

This is due to this part of the manifest:
```
	"type":"git",
	"url":"https://github.com/Subsurface-divelog/libdc.git",
	"branch": "Subsurface-branch",
	"commit": "c0f025b01920e5189a1e527a1ff995ca10c18d0e"
```

This breaks because the branch changed and is not on that commit anymore. You really only want to specify both when it is a tag which you want to ensure never changes. Either you want whatever is latest on a branch or you want a specific commit.

Also, the actual subsurface module just points to the latest master branch of git. That is not a good idea, it may randomly point to some broken thing the day that subsurface happens to be built in flathub. The goal of flathub is to ship stable releases, not random snapshots.

So, use release tags for the subsurface module, and pick the corresponding tags or commits for libdc that matches the given subsurface release.

**Comment by TingPing** [comment]
At least partially fixed by 287d624f3e455b5c94e5646d70a3c5f112a5aceb

**Comment by Alexander-Wilms** [comment]
Mh, I'm now using a release recent tarball, which requires no patches to build. But even though I'm using "-DCMAKE_BUILD_TYPE=Release" for the subsurface module, I still get this message when running it: "You are running a development version; latest version is 4.7.4.181.".

Regarding the googlemaps module: Their build script uses the master branch, regardless of whether one's building a development or release version: https://github.com/Subsurface-divelog/subsurface/blob/master/scripts/build.sh#L451
